src/resolvers/collaborator.ts

```diff
@@ -101,12 +123,136 @@ export const resolvers: Resolvers = {
       } catch (err) {
         if (err instanceof GraphQLError) throw err;
 
+        formatLogMessage(context).error(err, `Failure in ${reference}`);
+        throw InternalServerError();
+      }
+    },
+    // Add a collaborator to a Project
+    addProjectCollaborator: async (_, { projectId, email, userId }, context: MyContext): Promise<ProjectCollaborator> => {
+      const reference = 'addProjectCollaborator resolver';
+      try {
+        // if the user is an admin
+        if (isAdmin(context.token)) {
```

Comment:
Same as above. Collaborators with 'EDIT' or 'OWN' access on the project should be able to do this, so the `hasPermissionOnProject` check will be enough. No need to check if they are an admin